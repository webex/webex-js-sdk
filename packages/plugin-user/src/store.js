/**!
 *
 * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
 */

import {patterns} from '@ciscospark/common';

const usersByEmail = new WeakMap();
const usersById = new WeakMap();

export default class UserStore {
  constructor() {
    usersByEmail.set(this, new Map());
    usersById.set(this, new Map());
  }

  add(user) {
    if (!user.id) {
      return Promise.reject(`\`user.id\` is required`);
    }

    if (!user.emailAddress) {
      return Promise.reject(`\`user.emailAddress\` is required`);
    }

    const p1 = this.getById(user.id)
      .then((u) => usersById.get(this).set(user.id, Object.assign({}, u, user)))
      .catch(() => usersById.get(this).set(user.id, Object.assign({}, user)));

    const p2 = this.getByEmail(user.emailAddress)
      .then((u) => usersByEmail.get(this).set(user.emailAddress, Object.assign({}, u, user)))
      .catch(() => usersByEmail.get(this).set(user.emailAddress, Object.assign({}, user)));

    return Promise.all([p1, p2]);
  }

  get(id) {
    if (patterns.uuid.test(id)) {
      return this.getById(id);
    }

    if (patterns.email.test(id)) {
      return this.getByEmail(id);
    }

    return Promise.reject(`\`id\` does not appear to be a valid user identifier`);
  }

  getById(id) {
    const ret = usersById.get(this).get(id);
    if (ret) {
      return Promise.resolve(ret);
    }
    return Promise.reject(new Error(`No user found identified by specified identifier`));
  }

  getByEmail(email) {
    const ret = usersByEmail.get(this).get(email);
    if (ret) {
      return Promise.resolve(ret);
    }
    return Promise.reject(new Error(`No user found identified by specified identifier`));
  }
}
