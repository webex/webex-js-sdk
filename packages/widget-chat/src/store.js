import {applyMiddleware, combineReducers, compose, createStore} from 'redux';

import thunk from 'redux-thunk';
import createLogger from 'redux-logger';
import {user, conversation} from './reducers';
import sparkReducer from './modules/redux-spark/reducers';

const devtools = window.devToolsExtension || (() => (noop) => noop);

const enhancers = [
  applyMiddleware(
    thunk,
    createLogger({
      level: `info`,
      duration: true,
      collapsed: false,
      logErrors: process.env.NODE_ENV === `production`
    })
  ),
  devtools()
];

export default createStore(
  combineReducers({
    user,
    conversation,
    spark: sparkReducer
  }),
  compose(...enhancers)
);
