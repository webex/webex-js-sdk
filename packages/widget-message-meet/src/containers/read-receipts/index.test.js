import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';

import store from '../../store';

import ReadReceipts from '.';

describe(`ReadReceipts container`, () => {
  it(`renders properly`, () => {
    const component = renderer.create(
      <Provider store={store}>
        <ReadReceipts />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
