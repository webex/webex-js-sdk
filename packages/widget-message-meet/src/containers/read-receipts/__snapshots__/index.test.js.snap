exports[`ReadReceipts container renders properly 1`] = `<div />`;
